core/kotlinx-coroutines-test/src/TestCoroutineDispatcher.kt

```diff
@@ -0,0 +1,296 @@
+package kotlinx.coroutines.test
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.internal.ThreadSafeHeap
+import kotlinx.coroutines.test.internal.ThreadSafeHeapNode
+import java.util.concurrent.TimeUnit
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Control the virtual clock time of a [CoroutineDispatcher].
+ *
+ * Testing libraries may expose this interface to tests instead of [TestCoroutineDispatcher].
+ */
+@ExperimentalCoroutinesApi
+interface DelayController {
+    /**
+     * Returns the current virtual clock-time as it is known to this Dispatcher.
+     *
+     * @param unit The [TimeUnit] in which the clock-time must be returned.
+     * @return The virtual clock-time
+     */
+    @ExperimentalCoroutinesApi
+    fun currentTime(unit: TimeUnit = TimeUnit.MILLISECONDS): Long
+
+    /**
+     * Moves the Dispatcher's virtual clock forward by a specified amount of time.
+     *
+     * The amount the clock is progressed may be larger than the requested delayTime if the code under test uses
+     * blocking coroutines.
+     *
+     * @param delayTime The amount of time to move the CoroutineContext's clock forward.
+     * @param unit The [TimeUnit] in which [delayTime] and the return value is expressed.
```

Comment:
There are other coroutine methods (like `withTimeout`, `delay`) that just receive `timeMillis`. Even though I prefer an API that lets me pass `TimeUnit`, it might be better to stay consistent with existing APIs.

Reply:
Agreed, I am not sure I see a use case for this since delays will always be expressed in m. @elizarov is there a internal code path I don't know about that would trigger a non-ms delay?

Reply:
I've found this standard a little unexpected since Java API's will usually expect a `TimeUnit`, but as @yigit rightly says most Coroutine API's expect only a `timeMillis` param, so I'd opt for consistency also